projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionResultManager.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.execution;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import com.google.gson.Gson;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import com.vmware.taurus.service.KubernetesService;
+import com.vmware.taurus.service.model.ExecutionResult;
+import com.vmware.taurus.service.model.ExecutionStatus;
+import com.vmware.taurus.service.model.ExecutionTerminationStatus;
+
+@Slf4j
+public class JobExecutionResultManager {
+
+   static final String TERMINATION_MESSAGE_ATTRIBUTE_STATUS = "status";
+   static final String TERMINATION_MESSAGE_ATTRIBUTE_VDK_VERSION = "vdk_version";
+
+   @Data
+   @Builder
+   private static class TerminationMessage {
+      private String terminationStatus;
+      private String vdkVersion;
+   }
+
+   /**
+    * Extracts and determines the execution status, termination status and vdk version
+    * based on K8S Job status and K8S Pod termination message.
+    *
+    * @param jobExecution current job execution
+    * @return
+    */
+   public static ExecutionResult getResult(KubernetesService.JobExecution jobExecution) {
+      // Extracts attributes from JSON (e.g. status, vdk_version)
```

Comment:
Avoid adding comments in the code. I think it's better practice to add javadoc to the relevant method. This way other users of the method would see the same comment and does not clutter the code making it harder to read.